disnake/threads.py

```diff
@@ -959,3 +1010,254 @@ def _from_data(self, data: ThreadMemberPayload):
+        return PartialEmoji._from_name_id(self._emoji_name, self._emoji_id, state=self._state)
+
+    def to_dict(self) -> ThreadTagPayload:
+        payload: ThreadTagPayload = super().to_dict()  # type: ignore
+        payload["id"] = self.id
+        return payload
+
+    async def edit(
+        self,
+        *,
+        name: str = MISSING,
+        emoji: Optional[Union[str, Emoji, PartialEmoji]] = MISSING,
+        moderated: bool = MISSING,
+        reason: Optional[str] = None,
+    ) -> ThreadTag:
+        """|coro|
+
+        Edits the tag.
+
+        You must have the :attr:`~Permissions.manage_channels` permission to
+        do this.
+
+        Parameters
+        ----------
+        name: :class:`str`
+            The new tag name.
+        emoji: Optional[Union[:class:`str`, :class:`Emoji`, :class:`PartialEmoji`]]
+            The new tag emoji. Set to ``None`` to remove the emoji.
+        moderated: :class:`bool`
+            Whether only moderators can apply this tag to threads.
+        reason: Optional[:class:`str`]
+            The reason for editing the tag. Shows up on the audit log.
+
+        Raises
+        ------
+        InvalidData
+            Invalid channel/tag data was received from Discord.
+        Forbidden
+            You do not have permissions to edit the tag.
+        HTTPException
+            Editing the tag failed.
+
+        Returns
+        -------
+        :class:`ThreadTag`
+            The newly edited tag.
+        """
+        if isinstance(self._channel, Object):
+            raise TypeError("Cannot edit thread with no associated forum channel")
+
+        if self.id not in self._channel._available_tags:
+            # should only happen if tag was stored and tags were edited afterwards
+            raise ValueError("This tag is not part of the associated forum channel")
+
+        # we always have to patch the channel with the entire tag list,
+        # so serialize all tags and only replace the one matching `self.id`
+        new_tags = {tag_id: tag.to_dict() for tag_id, tag in self._channel._available_tags.items()}
+        # this always exists, see check above
+        new_tag_payload = new_tags[self.id]
+
+        if name is not MISSING:
+            new_tag_payload["name"] = name
+        if emoji is not MISSING:
+            emoji_name, emoji_id = PartialEmoji._to_name_id(emoji)
+            new_tag_payload["emoji_name"] = emoji_name
+            new_tag_payload["emoji_id"] = emoji_id
+        if moderated is not MISSING:
+            new_tag_payload["moderated"] = moderated
+
+        channel_data = cast(
+            "ForumChannelPayload",
+            await self._state.http.edit_channel(
+                self._channel.id, reason=reason, available_tags=list(new_tags.values())
+            ),
+        )
+        # TODO: update channel state here so that subsequent edits don't send old state?
```

Comment:
That *might* be a good idea, but we don't do that for other objects, so we might not want to start doing that here.

Reply:
Other objects generally don't need to send the full state on edit (only thing that comes to mind is `Guild.features`), but with how this endpoint works we always have to send all tags even if only one is being edited.

My main concern is this:
```py
await tag1.edit(name="new name 1")
await tag2.edit(name="new name 2")
```
If the `CHANNEL_UPDATE` for the first call isn't received+processed before the second call, the first edit will be reverted.

It's kinda similar to `Member.add_roles(atomic=)` which uses different endpoints depending on the `atomic` value, though we don't have that choice here.

Reply:
removed method, this is no longer relevant